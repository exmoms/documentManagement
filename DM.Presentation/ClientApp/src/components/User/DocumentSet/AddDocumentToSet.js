/**
 * @file Contains AddDocumentToSet component implementaion
 * @author Ali Daghman <ali.daghman@lit-co.net>
 * @version 1.0
 */

import React, { Component } from "react";
import DocumentTable from "../DocumentTable";
import SearchDocuments from "../SearchDocuments";
import CREATE_DOC_STYLES from "../Styles";
import { withStyles } from "@material-ui/core/styles";
import { withTranslation } from "react-i18next";
import Button from "@material-ui/core/Button";
import Dialog from "@material-ui/core/Dialog";
import DialogTitle from "@material-ui/core/DialogTitle";
import DialogContent from "@material-ui/core/DialogContent";
import DialogActions from "@material-ui/core/DialogActions";
import DialogContentText from "@material-ui/core/DialogContentText";

import TableContainer from "@material-ui/core/TableContainer";
import Table from "@material-ui/core/Table";
import TableRow from "@material-ui/core/TableRow";
import TableBody from "@material-ui/core/TableBody";
import TableCell from "@material-ui/core/TableCell";
import Paper from "@material-ui/core/Paper";

const styles = CREATE_DOC_STYLES;

class AddDocumentToSet extends Component {
  constructor(props) {
    super(props);
    this.state = {
      data: [],
      columns: [],
    };
    this.updateParentState = this.props.action;
  }

  getDocumentList = (list, columns) => {
    this.setState({
      data: list,
      columns: columns,
    });
  };

  handleClose = () => {
    this.setState(
      {
        data: [],
        columns: [],
      },
      this.updateParentState()
    );
  };

  /**
   * @todo Could we reduce rendering this component, once it renders two times
   * each time we update state
   */

  t = this.props["t"];
  render() {
    const { classes } = this.props;
    let lang = localStorage.getItem("i18nextLng");
    return (
      <div>
        <Dialog
          position="centerCenter"
          open={this.props.show}
          onClose={() => this.handleClose()}
          aria-labelledby="form-dialog-title"
        >
          <div
            style={
              lang === "ar" ? { textAlign: "right" } : { textAlign: "left" }
            }
          >
            <DialogTitle id="form-dialog-title">
              {this.t("documentset_addDocumentsToSet_title")}
            </DialogTitle>
          </div>
          <DialogContent>
            <div
              style={
                lang === "ar" ? { textAlign: "right" } : { textAlign: "left" }
              }
            >
              <DialogContentText>
                {this.t("documentset_addDocumentsToSet_Content")}
              </DialogContentText>
            </div>
            <TableContainer component={Paper}>
              <Table className={classes.table} aria-label="simple table">
                <TableBody>
                  <TableRow key="search-form">
                    <TableCell>
                      <SearchDocuments
                        getDocumentList={this.getDocumentList}
                        action="add-to-documentset"
                        setIdExclude={this.props.set.id}
                      />
                      <DocumentTable
                        data={this.state.data}
                        columns={this.state.columns}
                        set={this.props.set}
                        action="add-to-documentset"
                        handleClose={this.handleClose}
                      />
                    </TableCell>
                  </TableRow>
                </TableBody>
              </Table>
            </TableContainer>
          </DialogContent>
          <DialogActions>
            <Button
              id="CancelButton"
              onClick={() => this.handleClose()}
              color="primary"
            >
              {this.t("cancel")}
            </Button>
          </DialogActions>
        </Dialog>
      </div>
    );
  }
}

export default withStyles(styles)(withTranslation()(AddDocumentToSet));
