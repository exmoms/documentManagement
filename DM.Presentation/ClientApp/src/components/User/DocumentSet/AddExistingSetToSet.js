/**
 * @file Contains AddExistingSetToSet component implementaion
 * @author Ali Daghman <ali.daghman@lit-co.net>
 * @version 1.0
 */

import React, { Component, Fragment } from "react";
import { fetchData } from "../../../api/FetchData";
import { Redirect } from "react-router";
import { postDataToAPI } from "../../../api/PostData";
import { withStyles } from "@material-ui/core/styles";
import { withTranslation } from "react-i18next";
import Avatar from "@material-ui/core/Avatar";
import List from "@material-ui/core/List";
import ListItem from "@material-ui/core/ListItem";
import ListItemAvatar from "@material-ui/core/ListItemAvatar";
import ListItemText from "@material-ui/core/ListItemText";
import DialogTitle from "@material-ui/core/DialogTitle";
import Dialog from "@material-ui/core/Dialog";
import AddIcon from "@material-ui/icons/Add";
import { blue } from "@material-ui/core/colors";

const styles = {
  avatar: {
    backgroundColor: blue[100],
    color: blue[600],
  },
};

class AddExistingSetToSet extends Component {
  constructor(props) {
    super(props);
    this.state = {
      setId: 0,
      setsList: [],
      redirectToDocumentSet: false,
      openNotifiction: false,
      error: false,
      errorMessage: [],
    };
    this.updateParentState = this.props.action;
  }

  async componentDidMount() {
    // if (this.props.set.id !== this.state.setId && this.props.set.id !== 0) {
    let id = this.props.set.id;
    let url = `/api/DocumentSet/GetAllSetsExcludingSetsOfSet?Parent_documentSet_Id=${id}`;
    await fetchData(url)
      .then((value) => {
        this.setState({ setId: id, setsList: value });
      })
      .catch((e) => console.log(e));
    // }
  }

  componentDidUpdate() {
    if (this.state.redirectToDocumentSet) {
      this.setState(
        {
          redirectToDocumentSet: false,
        },
        this.handleClose()
      );
    }
  }

  addSetToSet = async (childSetId) => {
    let ParentSetId = this.props.set.id;

    let response = await postDataToAPI(
      `/api/DocumentSet/AddDocumentSetToDocumentSet?Parent_documentSet_Id=${ParentSetId}&child_documentSet_Id=${childSetId}`
    );

    if (response.ok) {
      this.setState(
        {
          redirectToDocumentSet: true,
          openNotifiction: true,
        } //,
        // this.handleClose()
      );
    } else {
      let json = await response.json();
      this.setState(
        {
          redirectToDocumentSet: true,
          name: "",
          openNotifiction: true,
          error: true,
          errorMessage: json.error,
        } //,
        //this.handleClose()
      );
    }
  };

  handleClose = () => {
    this.setState(
      {
        setId: 0,
        setsList: [],
        redirectToDocumentSet: false,
      },
      () => {
        this.updateParentState();
      }
    );
  };

  t = this.props["t"];

  render() {
    const { classes, show } = this.props;
    const { setsList } = this.state;
    let lang = localStorage.getItem("i18nextLng");
    return (
      <Fragment>
        <Dialog
          onClose={this.handleClose}
          aria-labelledby="simple-dialog-title"
          open={show}
          maxWidth={"xs"}
        >
          <div
            style={
              lang === "ar" ? { textAlign: "right" } : { textAlign: "left" }
            }
          >
            <DialogTitle id="simple-dialog-title">
              {this.t("documentset_addDocumentSetToSet_title")}
            </DialogTitle>
          </div>
          {setsList.length > 0 ? (
            <List>
              {setsList.map((set, index) => {
                return (
                  <ListItem
                    key={index}
                    button
                    onClick={() => this.addSetToSet(set.id)}
                  >
                    <ListItemAvatar>
                      <Avatar className={classes.avatar}>
                        <AddIcon />
                      </Avatar>
                    </ListItemAvatar>
                    <ListItemText primary={set.name} />
                  </ListItem>
                );
              })}
            </List>
          ) : (
            <List>
              <ListItem button onClick={() => this.handleClose()}>
                <ListItemAvatar>
                  <Avatar className={classes.avatar}>
                    <AddIcon />
                  </Avatar>
                </ListItemAvatar>
                <ListItemText primary={this.t("documentset_noSetFound")} />
              </ListItem>
            </List>
          )}
        </Dialog>
        {this.state.redirectToDocumentSet && (
          <Redirect
            to={{
              pathname: "/document-set",
              state: {
                id: this.props.set.id,
                name: this.props.set.name,
                openNotifiction: this.state.openNotifiction,
                error: this.state.error,
                errorMessage: this.state.errorMessage,
              },
            }}
          />
        )}
      </Fragment>
    );
  }
}

export default withStyles(styles)(withTranslation()(AddExistingSetToSet));
