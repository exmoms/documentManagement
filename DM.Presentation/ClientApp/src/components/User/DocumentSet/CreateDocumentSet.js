/**
 * @file Contains CreateDocumentSet component implementaion
 * @author Ali Daghman <ali.daghman@lit-co.net>
 * @version 1.0
 */

import React, { Fragment } from "react";
import { Redirect } from "react-router";
import { TextField } from "@material-ui/core";
import { withStyles } from "@material-ui/core/styles";
import Button from "@material-ui/core/Button";
import { postDataToAPI } from "../../../api/PostData";
import { withTranslation } from "react-i18next";
import Dialog from "@material-ui/core/Dialog";
import DialogTitle from "@material-ui/core/DialogTitle";
import DialogContent from "@material-ui/core/DialogContent";
import DialogActions from "@material-ui/core/DialogActions";

const styles = (theme) => ({
  root: {
    marginTop: theme.spacing(8),
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
  },
  box: {
    padding: "10px",
  },
  textField: {
    marginLeft: theme.spacing(1),
    marginRight: theme.spacing(1),
    width: 300,
  },
  Button: {
    marginTop: "70px",
  },
});

class CreateDocumentSet extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      name: "",
      redirectToDocumentSet: false,
      openNotifiction: false,
      error: false,
      errorMessage: [],
    };

    this.handleClose = this.props.action;
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  componentDidUpdate() {
    if (this.state.redirectToDocumentSet) {
      this.setState(
        {
          redirectToDocumentSet: false,
        },
        this.handleClose()
      );
    }
  }

  handleChange(event) {
    this.setState({ name: event.target.value });
  }

  handleSubmit = async (event) => {
    let set = this.props.set;
    let response = {};
    if (set.id === 0) {
      response = await postDataToAPI("/api/DocumentSet/AddNewDocumentSet", {
        Name: this.state.name,
      });
    } else {
      response = await postDataToAPI("/api/DocumentSet/AddNewDocumentSet", {
        Name: this.state.name,
        ParentDocumentSet: set.id,
      });
    }
    if (response.ok) {
      this.setState(
        {
          redirectToDocumentSet: true,
          name: "",
          openNotifiction: true,
        } //,
        //this.handleClose()
      );
    } else {
      let json = await response.json();
      this.setState(
        {
          redirectToDocumentSet: true,
          name: "",
          openNotifiction: true,
          error: true,
          errorMessage: json.error,
        } //,
        // this.handleClose()
      );
    }
  };

  t = this.props["t"];
  render() {
    const { classes } = this.props;
    let lang = localStorage.getItem("i18nextLng");
    return (
      <Fragment>
        <Dialog
          position="centerCenter"
          open={this.props.show}
          onClose={() => this.handleClose()}
          maxWidth={"xs"}
        >
          <div
            style={
              lang === "ar" ? { textAlign: "right" } : { textAlign: "left" }
            }
          >
            <DialogTitle id="simple-dialog-title">
              {this.t("documentset_addnewdocumentset")}
            </DialogTitle>
          </div>
          <DialogContent>
            <div className={classes.root}>
              <div className={classes.box}>
                <div
                  style={
                    lang === "ar"
                      ? { textAlign: "right" }
                      : { textAlign: "left" }
                  }
                >
                  {this.t("documentset_entername")}
                </div>
                <TextField
                  id="standard-basic"
                  className={classes.textField}
                  required
                  helperText={this.t("hypertext_required")}
                  label={this.t("namelabel")}
                  fullWidth
                  value={this.state.name}
                  onChange={this.handleChange}
                />
                <br />
                <Button
                  variant="contained"
                  className={classes.Button}
                  color="primary"
                  fullWidth
                  onClick={this.handleSubmit}
                >
                  {" "}
                  {this.t("button_submit")}
                </Button>
              </div>
            </div>
          </DialogContent>
          <DialogActions>
            <Button
              id="CancelButton"
              onClick={() => this.handleClose()}
              color="primary"
            >
              {this.t("cancel")}
            </Button>
          </DialogActions>
        </Dialog>
        {this.state.redirectToDocumentSet && (
          <Redirect
            to={{
              pathname: "/document-set",
              state: {
                id: this.props.set.id,
                name: this.props.set.name,
                openNotifiction: this.state.openNotifiction,
                error: this.state.error,
                errorMessage: this.state.errorMessage,
              },
            }}
          />
        )}
      </Fragment>
    );
  }
}

export default withStyles(styles, { withTheme: true })(
  withTranslation()(CreateDocumentSet)
);
