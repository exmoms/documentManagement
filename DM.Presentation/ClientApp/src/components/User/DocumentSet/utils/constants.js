const FILE = 0;
const FOLDER = 1;

export { FILE, FOLDER };
