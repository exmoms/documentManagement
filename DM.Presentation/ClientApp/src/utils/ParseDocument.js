/**
 * This function is responsible of parsing document data (from api/document/getDocumentVersionById?id={id}) into a form readable
 * By react-jsonschema-form library
 * @param {Object}  res : the document as json object
 * @param {Number} option : its value takes 0 for preview document, otherwise for update document
 * @return {Object} consist of four parts:
 *      1. schema: the schema of the form expected by react-jsonschema-form library
 *      2. ui_schema: the wedgit corresponding to each form field.
 *      3. form_data: the form fields data
 *      3. map:  maping attributeName and corresponding attributeId
 */

function parseDocumentDataToForm(res, option) {
  let schema = {
    title: "",
    type: "object",
    required: [],
    properties: {
      metadataModelName: { type: "string", title: "Metadata Model Name" },
      addedDate: { type: "string", title: " Document Added Date" },
      latestVersion: { type: "number", title: "Latest Version" },
      versionMessage: { type: "string", title: "Version Message" },
      documentVersionAddedDate: {
        type: "string",
        title: "Document Version Added Date",
      },
    },
  };

  let form_data = {
    metadataModelName: res.metadataModelName,
    addedDate: res.addedDate.substr(0, 10),
    latestVersion: res.latestVersion,
    versionMessage: res.documentVersion.versionMessage,
    documentVersionAddedDate: res.documentVersion.addedDate.substr(0, 10),
  };

  let ui_schema = {
    metadataModelName: {
      "ui:readonly": true,
    },
    addedDate: {
      "ui:readonly": true,
    },
    latestVersion: {
      "ui:readonly": true,
    },
    versionMessage: {
      "ui:readonly": true,
    },
    documentVersionAddedDate: {
      "ui:readonly": true,
    },
  };

  let map = [];
  schema.title = res.name;

  // if option is update we provide a required "Document Version" field.
  if (option === 1) {
    schema.required.push("documentVersion");
    Object.defineProperty(schema.properties, "documentVersion", {
      value: {
        type: "string",
        title: "New Version Message",
      },
      writable: true,
      enumerable: true,
      configurable: true,
    });
  }
  // loop over all metadata attributes in the documents
  res.documentVersion.values.forEach((element) => {
    map[element.attributeName] = element.attributeId;
    if (element.isRequired) {
      schema.required.push(element.attributeName);
    }
    const name = element.attributeName;

    // if the option = 0 then we should add readonly property to the ui_schema for this field..
    if (option === 0) {
      Object.defineProperty(ui_schema, name, {
        value: { "ui:readonly": true },
        writable: true,
        enumerable: true,
        configurable: true,
      });
    }

    Object.defineProperty(schema.properties, name, {
      value: {
        type: "",
        title: name,
      },
      writable: true,
      enumerable: true,
      configurable: true,
    });

    switch (element.typeId) {
      case 1:
        schema.properties[name].type = "boolean";
        break;
      case 2:
        schema.properties[name].type = "string";
        if (ui_schema.hasOwnProperty(name)) {
          Object.defineProperty(ui_schema, name, {
            value: {
              "ui:widget": "date",
              "ui:readonly": true,
            },
            writable: true,
            enumerable: true,
            configurable: true,
          });
        } else {
          Object.defineProperty(ui_schema, name, {
            value: {
              "ui:widget": "date",
            },
            writable: true,
            enumerable: true,
            configurable: true,
          });
        }

        break;
      case 3:
      case 4:
      case 5:
        schema.properties[name].type = "number";
        break;
      case 6:
        schema.properties[name].type = "string";
        break;
      default:
      //@ todo: we should return error code to indecate unsupported types.
    }

    Object.defineProperty(form_data, name, {
      value: element.typeId !== 2 ? element.value : element.value.substr(0, 10),
      writable: true,
      enumerable: true,
      configurable: true,
    });
  });

  return { schema, ui_schema, form_data, map };
}

export default parseDocumentDataToForm;
