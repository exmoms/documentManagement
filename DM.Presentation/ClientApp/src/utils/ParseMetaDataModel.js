/**
 * This function is responsible of parsing metadata model data (from api/document/getDocumentVersionById?id={id}) into a form readable
 * By react-jsonschema-form library
 * @param {Object}  res : the metadata Model as json form
 * @return {Object} consist of three parts:
 *      1. json_schema: the schema of the form expected by react-jsonschema-form library
 *      2. ui_schema: the wedgit corresponding to each form field.
 *      3. map:  maping metaDataAttributeName and corresponding metaDataAttributeId
 */

function parseMetaDataModel(res) {
  let json_schema = {
    title: "",
    type: "object",
    required: [],
    properties: {},
  };

  let ui_schema = {};

  let map = [];
  json_schema.title = res.metaDataModelName;
  res.metaDataAttributes.forEach((element) => {
    map[element.metaDataAttributeName] = element.id;
    if (element.isRequired) {
      json_schema.required.push(element.metaDataAttributeName);
    }
    const name = element.metaDataAttributeName;
    Object.defineProperty(json_schema.properties, name, {
      value: {
        type: "",
        title: "",
        default: "",
      },
      writable: true,
      enumerable: true,
      configurable: true,
    });

    json_schema.properties[name].title = name;
    const type = element.dataTypeID;
    if (type === 6) {
      json_schema.properties[name].type = "string";
    } else if (type === 3 || type === 5 || type === 4) {
      json_schema.properties[name].type = "number";
    } else if (type === 1) {
      json_schema.properties[name].type = "boolean";
      json_schema.properties[name].default = false;
    } else if (type === 2) {
      json_schema.properties[name].type = "string";
      Object.defineProperty(ui_schema, name, {
        value: {
          "ui:widget": "date",
        },
        writable: true,
        enumerable: true,
        configurable: true,
      });
    } else {
      //@ todo: we should return error code to indecate unsupported types.
    }
  });
  return { json_schema, ui_schema, map };
}

export default parseMetaDataModel;
