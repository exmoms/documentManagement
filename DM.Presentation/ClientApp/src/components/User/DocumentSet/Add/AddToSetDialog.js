/**
 * @file Contains AddToSetDialog component implementaion
 * @author Ali Daghman <ali.daghman@lit-co.net>
 * @version 1.0
 */

import React, { Fragment } from "react";
import { makeStyles } from "@material-ui/core/styles";
import Avatar from "@material-ui/core/Avatar";
import List from "@material-ui/core/List";
import ListItem from "@material-ui/core/ListItem";
import ListItemAvatar from "@material-ui/core/ListItemAvatar";
import ListItemText from "@material-ui/core/ListItemText";
import DialogTitle from "@material-ui/core/DialogTitle";
import Dialog from "@material-ui/core/Dialog";
import AddIcon from "@material-ui/icons/Add";
import { blue } from "@material-ui/core/colors";
import { FOLDER, FILE } from "../utils/constants";
import { useTranslation } from "react-i18next";

const useStyles = makeStyles({
  avatar: {
    backgroundColor: blue[100],
    color: blue[600],
  },
});

export default function AddToSetDialog(props) {
  const classes = useStyles();
  const { show, set } = props;

  const handleClose = () => {
    props.action();
  };

  const { t } = useTranslation();
  let lang = localStorage.getItem("i18nextLng");
  return (
    <Dialog
      onClose={handleClose}
      aria-labelledby="simple-dialog-title"
      open={show}
      maxWidth={"xs"}
    >
      <div
        style={lang === "ar" ? { textAlign: "right" } : { textAlign: "left" }}
      >
        <DialogTitle id="simple-dialog-title">{t("actions")}</DialogTitle>
      </div>
      <List>
        <ListItem autoFocus button onClick={() => props.action(FOLDER, true)}>
          <ListItemAvatar>
            <Avatar className={classes.avatar}>
              <AddIcon />
            </Avatar>
          </ListItemAvatar>
          <ListItemText primary={t("documentset_addnewdocumentset")} />
        </ListItem>

        {set.id !== 0 ? (
          <Fragment>
            <ListItem
              autoFocus
              button
              onClick={() => props.action(FOLDER, false)}
            >
              <ListItemAvatar>
                <Avatar className={classes.avatar}>
                  <AddIcon />
                </Avatar>
              </ListItemAvatar>
              <ListItemText primary={t("add_existing_set")} />
            </ListItem>
            <ListItem autoFocus button onClick={() => props.action(FILE)}>
              <ListItemAvatar>
                <Avatar className={classes.avatar}>
                  <AddIcon />
                </Avatar>
              </ListItemAvatar>
              <ListItemText primary={t("add_document_to_set")} />
            </ListItem>
          </Fragment>
        ) : (
          ""
        )}
      </List>
    </Dialog>
  );
}
