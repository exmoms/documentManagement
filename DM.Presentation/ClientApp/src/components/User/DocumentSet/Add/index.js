/**
 * @file Contains Add component implementaion
 * @author Ali Daghman <ali.daghman@lit-co.net>
 * @version 1.0
 */

import React, { Fragment, Component } from "react";
import styled from "styled-components";
import AddToSetDialog from "./AddToSetDialog";
import AddDocumentToSet from "../AddDocumentToSet";
import AddExistingSetToSet from "../AddExistingSetToSet";
import CreateDocumentSet from "../CreateDocumentSet";
import { FOLDER, FILE } from "../utils/constants";

class Add extends Component {
  constructor(props) {
    super(props);

    this.state = {
      open: false,
      showAddDocument: false,
      showAddNewSet: false,
      showAddExistingSet: false,
    };
  }

  handleOpen = () => {
    this.setState({
      open: true,
      showAddDocument: false,
      showAddNewSet: false,
      showAddExistingSet: false,
    });
  };

  handleChoice = (type, isNew) => {
    if (type === FOLDER) {
      if (isNew) {
        this.setState({
          open: false,
          showAddNewSet: true,
          showAddDocument: false,
          showAddExistingSet: false,
        });
      } else {
        this.setState({
          open: false,
          showAddNewSet: false,
          showAddDocument: false,
          showAddExistingSet: true,
        });
      }
    } else if (type === FILE) {
      this.setState({
        open: false,
        showAddDocument: true,
        showAddNewSet: false,
      });
    } else {
      this.setState({
        open: false,
        showAddDocument: false,
        showAddNewSet: false,
        showAddExistingSet: false,
      });
    }
  };

  closeAddDocument = () => {
    this.setState({
      showAddDocument: false,
    });
  };

  closeAddNewSet = () => {
    this.setState({
      showAddNewSet: false,
    });
  };

  closeAddExistingSet = () => {
    this.setState({
      showAddExistingSet: false,
    });
  };

  render() {
    return (
      <Fragment>
        <Container onClick={() => this.handleOpen()}>+</Container>
        <AddToSetDialog
          set={this.props.set}
          show={this.state.open}
          action={this.handleChoice}
        />

        {this.state.showAddDocument && (
          <AddDocumentToSet
            set={this.props.set}
            show={this.state.showAddDocument}
            action={this.closeAddDocument}
          />
        )}

        {this.state.showAddExistingSet && (
          <AddExistingSetToSet
            set={this.props.set}
            show={this.state.showAddExistingSet}
            action={this.closeAddExistingSet}
          />
        )}

        {this.state.showAddNewSet && (
          <CreateDocumentSet
            set={this.props.set}
            show={this.state.showAddNewSet}
            action={this.closeAddNewSet}
          />
        )}
      </Fragment>
    );
  }
}

export default Add;

const Container = styled.div`
  height: 109px;
  width: 96px;
  border: 3px dashed #dee0e4;
  display: flex;
  justify-content: center;
  border-radius: 8px;
  align-items: center;
  font-size: 30px;
  color: #dee0e4;
  margin: -6px 21px;
  cursor: copy;
`;
