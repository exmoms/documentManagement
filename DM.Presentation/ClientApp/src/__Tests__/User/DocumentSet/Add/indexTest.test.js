import React from "react";
import ReactDom from "react-dom";
import Add from "../../../../components/User/DocumentSet/Add/index";
import { cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import { act } from "react-dom/test-utils";
import { mount } from "enzyme";
import { createMount } from '@material-ui/core/test-utils';
import Dialog from "@material-ui/core/Dialog";
import { Fragment } from "react";
import ListItem from "@material-ui/core/ListItem";
import { FOLDER, FILE } from "../../../../components/User/DocumentSet/utils/constants";
import AddToSetDialog from "../../../../components/User/DocumentSet/Add/AddToSetDialog";
import AddDocumentToSet from "../../../../components/User/DocumentSet/AddDocumentToSet";
import AddExistingSetToSet from "../../../../components/User/DocumentSet/AddExistingSetToSet";
import CreateDocumentSet from "../../../../components/User/DocumentSet/CreateDocumentSet";
afterEach(cleanup);

describe("index", () => {

    test("renders without crashing", () => {
        let set={id:1};
        const div = document.createElement("div");
        ReactDom.render(<Add set={set}/>, div);
      });

    test("index matches snapshot", () => {
        let set={id:1};
    const renderedValue = createMount()(
        <Add set={set}/>
      );
      expect(renderedValue.html()).toMatchSnapshot();
    });
   
    test("index test intialization", () => {
        let set={id:1};
        const wrapper = mount(
            <Add set={set}/>
          );
          expect(wrapper.find(AddToSetDialog).length).toBe(1);
          expect(wrapper.find(AddToSetDialog).at(0).props().set).toStrictEqual(set);
          expect(wrapper.find(AddToSetDialog).at(0).props().show).toBeFalsy();

          expect(wrapper.find(AddDocumentToSet).length).toBe(0);

          expect(wrapper.find(AddExistingSetToSet).length).toBe(0);

          expect(wrapper.find(CreateDocumentSet).length).toBe(0);
      });

      test("handleChoice when (type === FOLDER) and isNew", () => {
        let set={id:1};
        let type=FOLDER;
        let isNew=true;
        const wrapper = mount(
            <Add set={set}/>
          );

          act(() => {
            wrapper
              .find(AddToSetDialog)
              .at(0)
              .props()
              .action(type,isNew);
          });
          wrapper.render();
          wrapper.update();
          expect(wrapper.find(AddToSetDialog).length).toBe(1);
          expect(wrapper.find(AddToSetDialog).at(0).props().set).toStrictEqual(set);
          expect(wrapper.find(AddToSetDialog).at(0).props().show).toBeFalsy();

          expect(wrapper.find(AddDocumentToSet).length).toBe(0);

          expect(wrapper.find(AddExistingSetToSet).length).toBe(0);

          expect(wrapper.find(CreateDocumentSet).length).toBe(1);
          expect(wrapper.find(CreateDocumentSet).at(0).props().set).toStrictEqual(set);
          expect(wrapper.find(CreateDocumentSet).at(0).props().show).toBeTruthy();
      });

      test("handleChoice when (type === FOLDER) and isNew == false", () => {
        let set={id:1};
        let type=FOLDER;
        let isNew=false;
        const wrapper = mount(
            <Add set={set}/>
          );

          act(() => {
            wrapper
              .find(AddToSetDialog)
              .at(0)
              .props()
              .action(type,isNew);
          });
          wrapper.render();
          wrapper.update();
          expect(wrapper.find(AddToSetDialog).length).toBe(1);
          expect(wrapper.find(AddToSetDialog).at(0).props().set).toStrictEqual(set);
          expect(wrapper.find(AddToSetDialog).at(0).props().show).toBeFalsy();

          expect(wrapper.find(AddDocumentToSet).length).toBe(0);

          expect(wrapper.find(AddExistingSetToSet).length).toBe(1);
          expect(wrapper.find(AddExistingSetToSet).at(0).props().set).toStrictEqual(set);
          expect(wrapper.find(AddExistingSetToSet).at(0).props().show).toBeTruthy();

          expect(wrapper.find(CreateDocumentSet).length).toBe(0);
      });
     
      test("handleChoice when (type === FILE)", () => {
        let set={id:1};
        let type=FILE;
        let isNew=false;
        const wrapper = mount(
            <Add set={set}/>
          );

          act(() => {
            wrapper
              .find(AddToSetDialog)
              .at(0)
              .props()
              .action(type,isNew);
          });
          wrapper.render();
          wrapper.update();
          expect(wrapper.find(AddToSetDialog).length).toBe(1);
          expect(wrapper.find(AddToSetDialog).at(0).props().set).toStrictEqual(set);
          expect(wrapper.find(AddToSetDialog).at(0).props().show).toBeFalsy();

          expect(wrapper.find(AddDocumentToSet).length).toBe(1);
          expect(wrapper.find(AddDocumentToSet).at(0).props().set).toStrictEqual(set);
          expect(wrapper.find(AddDocumentToSet).at(0).props().show).toBeTruthy();

          expect(wrapper.find(AddExistingSetToSet).length).toBe(0);

          expect(wrapper.find(CreateDocumentSet).length).toBe(0);
      });
     
      test("handleChoice when (type is not FOLDER , FILE )", () => {
        let set={id:1};
        let type="";
        let isNew=false;
        const wrapper = mount(
            <Add set={set}/>
          );

          act(() => {
            wrapper
              .find(AddToSetDialog)
              .at(0)
              .props()
              .action(type,isNew);
          });
          wrapper.render();
          wrapper.update();
          expect(wrapper.find(AddToSetDialog).length).toBe(1);
          expect(wrapper.find(AddToSetDialog).at(0).props().set).toStrictEqual(set);
          expect(wrapper.find(AddToSetDialog).at(0).props().show).toBeFalsy();

          expect(wrapper.find(AddDocumentToSet).length).toBe(0);

          expect(wrapper.find(AddExistingSetToSet).length).toBe(0);

          expect(wrapper.find(CreateDocumentSet).length).toBe(0);
      });
    
      test("closeAddDocument", () => {
        let set={id:1};
        let type=FILE;
        let isNew=false;
        const wrapper = mount(
            <Add set={set}/>
          );

          act(() => {
            wrapper
              .find(AddToSetDialog)
              .at(0)
              .props()
              .action(type,isNew);
          });
          wrapper.render();
          wrapper.update();
          expect(wrapper.find(AddToSetDialog).length).toBe(1);
          expect(wrapper.find(AddToSetDialog).at(0).props().set).toStrictEqual(set);
          expect(wrapper.find(AddToSetDialog).at(0).props().show).toBeFalsy();

          expect(wrapper.find(AddDocumentToSet).length).toBe(1);
          expect(wrapper.find(AddDocumentToSet).at(0).props().set).toStrictEqual(set);
          expect(wrapper.find(AddDocumentToSet).at(0).props().show).toBeTruthy();

          expect(wrapper.find(AddExistingSetToSet).length).toBe(0);

          expect(wrapper.find(CreateDocumentSet).length).toBe(0);
          act(() => {
            wrapper
              .find(AddDocumentToSet)
              .at(0)
              .props()
              .action();
          });
          wrapper.render();
          wrapper.update();

          expect(wrapper.find(AddDocumentToSet).length).toBe(0);
      });
      
      test("closeAddExistingSet", () => {
        let set={id:1};
        let type=FOLDER;
        let isNew=false;
        const wrapper = mount(
            <Add set={set}/>
          );

          act(() => {
            wrapper
              .find(AddToSetDialog)
              .at(0)
              .props()
              .action(type,isNew);
          });
          wrapper.render();
          wrapper.update();
          expect(wrapper.find(AddToSetDialog).length).toBe(1);
          expect(wrapper.find(AddToSetDialog).at(0).props().set).toStrictEqual(set);
          expect(wrapper.find(AddToSetDialog).at(0).props().show).toBeFalsy();

          expect(wrapper.find(AddDocumentToSet).length).toBe(0);

          expect(wrapper.find(AddExistingSetToSet).length).toBe(1);
          expect(wrapper.find(AddExistingSetToSet).at(0).props().set).toStrictEqual(set);
          expect(wrapper.find(AddExistingSetToSet).at(0).props().show).toBeTruthy();

          expect(wrapper.find(CreateDocumentSet).length).toBe(0);

          act(() => {
            wrapper
              .find(AddExistingSetToSet)
              .at(0)
              .props()
              .action();
          });
          wrapper.render();
          wrapper.update();

          expect(wrapper.find(AddExistingSetToSet).length).toBe(0);
      });
     
      test("closeAddNewSet", () => {
        let set={id:1};
        let type=FOLDER;
        let isNew=true;
        const wrapper = mount(
            <Add set={set}/>
          );

          act(() => {
            wrapper
              .find(AddToSetDialog)
              .at(0)
              .props()
              .action(type,isNew);
          });
          wrapper.render();
          wrapper.update();
          expect(wrapper.find(AddToSetDialog).length).toBe(1);
          expect(wrapper.find(AddToSetDialog).at(0).props().set).toStrictEqual(set);
          expect(wrapper.find(AddToSetDialog).at(0).props().show).toBeFalsy();

          expect(wrapper.find(AddDocumentToSet).length).toBe(0);

          expect(wrapper.find(AddExistingSetToSet).length).toBe(0);

          expect(wrapper.find(CreateDocumentSet).length).toBe(1);
          expect(wrapper.find(CreateDocumentSet).at(0).props().set).toStrictEqual(set);
          expect(wrapper.find(CreateDocumentSet).at(0).props().show).toBeTruthy();

          act(() => {
            wrapper
              .find(CreateDocumentSet)
              .at(0)
              .props()
              .action();
          });
          wrapper.render();
          wrapper.update();
          expect(wrapper.find(CreateDocumentSet).length).toBe(0);
      });

});