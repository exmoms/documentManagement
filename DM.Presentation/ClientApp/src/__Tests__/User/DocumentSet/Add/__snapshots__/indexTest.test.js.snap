// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index index matches snapshot 1`] = `"<div class=\\"sc-AxhUy feYMlc\\">+</div>"`;
